### ISSUE TITLE
Buffer returned by the signMessage method is not acting like a Buffer object

### ISSUE THREAD

**Issue by earrietadev** [stellar/freighter#1626]
### What version are you using?
Wallet version: 5.24.1
Freighter CLI version: 3.0.0

### What did you do?

When using the method `signMessage` the [code](https://github.com/stellar/freighter/blob/3e1c50a118ea6ad5faa2eae8d336df82be855d74/%40stellar/freighter-api/src/signMessage.ts#L13) says the it will return a `Buffer` (I assume is the NodeJS Buffer object).

But because I'm following [SEP-0043](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0043.md#wallet-interface-format) I'm converting that response to a `string`... But the method `toString` is not acting like the NodeJS Buffer so I feel is not a Buffer the response.

### What did you expect to see?
A `Buffer` object with the features the NodeJS `Buffer` has

### What did you see instead?

The response was an `Object` with the structure:

```ts
{
  data: number[]
  type: "Buffer"
}
```

**Comment by aristidesstaffieri** [comment]
👋  so what we do for Buffer on the browser side is use a webpack polyfill which uses https://github.com/feross/buffer for any Buffer references, or at least it should.

After reviewing the build steps, it looks like we polyfill Buffer in the extension build but not in the freighter-api build, so on the extension side we get a reference to [this buffer class](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/902de1e58fa435ec9d6eac19f2072ba5be2d8d36/types/node/v16/buffer.d.ts#L296) which should be the class you expect but on the API side we resolve buffer to [this interface](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/902de1e58fa435ec9d6eac19f2072ba5be2d8d36/types/node/buffer.buffer.d.ts#L286).

Thanks for finding this, I'm going to try to polyfill it the same way on both builds and will test this API again and get back to you here with details.

**Comment by aristidesstaffieri** [comment]
@earrietadev @piyalbasu the spec says that this should actually return - 
```
Promise<{
      signedAuthEntry: string;
      signerAddress: string;
    } & error?: Error>
```

so I think we need to make a breaking change and return a string instead of a buffer to match the spec. Any objection to making this change? @earrietadev it seemed like in your original issue you point out that our Buffer implementation might not be polyfilling the expected API, I think we can ignore this and make the breaking change if we agree that this is the correct API for `signMessage`.

**Comment by earrietadev** [comment]
@aristidesstaffieri yeah I'm totally ok with that, I was converting the Buffer to a string to follow SEP-0043 and I implemented it in a way that once Freighter got updated to follow SEP-0043 it shouldn't break the wallets kit integration so no issues on our end

**Comment by aristidesstaffieri** [comment]
@earrietadev [here's what I came up with](https://github.com/stellar/freighter/pull/1755) to patch this. New versions will return a b64 encoded string so you can verify like -

```
 const kp = <signing key pair>
 const res = await stellarApi.signMessage("hi", { networkPassphrase: SorobanClient.Networks.TESTNET })
 kp.verify(Buffer.from("hi", "base64"), Buffer.from(res.signedMessage))
```

**Comment by aristidesstaffieri** [comment]
@earrietadev this is now live in version 4 of the API.
https://www.npmjs.com/package/@stellar/freighter-api?activeTab=code

Let me know if you have any questions.
